### PR TITLE
Add the missing entries to the release notes 1.5 and 1.6

### PR DESCRIPTION
**Pull request by maelvls**
- I had to use `##` (section level 2) instead of `#` (section level 1) because the section level 1 is meant for titles and does not show in the table of contents.
- I reviewed all the change log entries and rephrased the ones that I could not understand. I also updated the `release-note` block for those PRs.

### CODE REVIEW

**Review by jakexks — APPROVED** [review]
/lgtm

File: content/en/docs/release-notes/release-notes-1.7.md
Comment (wallrj) [inline content/en/docs/release-notes/release-notes-1.7.md:132]: v1.6.1 was correct. That's the tag I used when running the release-notes tool. 
Or perhaps it should say  `Changelog since release 1.6` :shrug:

Comment (maelvls) [reply content/en/docs/release-notes/release-notes-1.7.md:132]: I thought the `START_REV` would need to be `v1.6.0` when releasing `v1.7.0`, but I now realize that the release process doesn't make it clear whether we should be using the latest patch version from the previous branch (e.g. v1.2.1 in the below screenshot) or the "first" release from the previous branch (v1.2.0 in the below screenshot):

Screenshot from 2022-01-30 15-35-25

The reason I changed this to `v1.6.0` is that (I think) people are expecting to see all the changes that happened from the moment we released the last major release (that moment being when we release v1.6.0).

Since v1.6.1 is on a "separate lifeline", we shouldn't include those commits (although the release notes for those

 commits will show since the commits that made `v1.6.1` are cherry-picked).

Just to check, I tried both ways:

```sh
START_REV="v1.6.0" END_REV="release-1.7" BRANCH="release-1.7" RELEASE_VERSION="1.7.0" release-notes --repo-path=cert-manager --org=jetstack --repo=cert-manager --required-author=jetstack-bot --output release-notes-1.6.0.md
START_REV="v1.6.1" END_REV="release-1.7" BRANCH="release-1.7" RELEASE_VERSION="1.7.0" release-notes --repo-path=cert-manager --org=jetstack --repo=cert-manager --required-author=jetstack-bot --output release-notes-1.6.1.md
```

Results: [release-notes-1.6.0.md](https://github.com/cert-manager/website/files/7966110/release-notes-1.6.0.md), [release-notes-1.6.1.md](https://github.com/cert-manager/website/files/7966111/release-notes-1.6.1.md)

Using `START_REV=v1.6.1`, some change entries are "skipped" compared to `START_REV=v1.6.0`:

```diff
--- release-notes-1.6.0.md	2022-01-30 15:34:16.001618221 +0100
+++ release-notes-1.6.1.md	2022-01-30 15:33:49.724916679 +0100
@@ -1,7 +1,7 @@
 # Release notes for release-1.7
 
 [Documentation](https://docs.k8s.io/docs/home)
-# Changelog since v1.6.0
+# Changelog since v1.6.1
 
 ## Changes by Kind
 
@@ -14,8 +14,6 @@
 - Added `additionalOutputFormats` parameter to allow `DER` (binary) and `CombinedPEM` (key + cert bundle) formats. ([#4598](https://github.com/jetstack/cert-manager/pull/4598), [@seuf](https://github.com/seuf))
 - Added a makefile based build workflow which doesn't depend on bazel ([#4554](https://github.com/jetstack/cert-manager/pull/4554), [@SgtCoDFish](https://github.com/SgtCoDFish))
 - Added a new Helm chart parameter `prometheus.servicemonitor.honorLabels`, which sets the `honor_labels` field  of the Prometheus scrape config. ([#4608](https://github.com/jetstack/cert-manager/pull/4608), [@thirdeyenick](https://github.com/thirdeyenick))
-- Adds release targets for both `cmctl` as well as `kubectl-cert_manager` ([#4523](https://github.com/jetstack/cert-manager/pull/4523), [@JoshVanL](https://github.com/JoshVanL))
-- Breaking change: pprof now runs by default on localhost:6060 for webhook and controller, but only if explicitly enabled. Pprof can now be enabled also for cainjector. All three components have `--enable-profiling`, `--profiler-address` CLI flags to configure profiling. Thanks to @bitscuit for help with this! ([#4550](https://github.com/jetstack/cert-manager/pull/4550), [@irbekrm](https://github.com/irbekrm))
 - Certificate Secrets are now managed by the APPLY API call, rather than UPDATE/CREATE. The issuing controller actively reconciles Certificate SecretTemplate's against corresponding Secrets, garbage collecting and correcting key/value changes. ([#4638](https://github.com/jetstack/cert-manager/pull/4638), [@JoshVanL](https://github.com/JoshVanL))
 
 ### Bug or Regression
@@ -25,18 +23,14 @@
 - Fixed a bug in the way the Helm chart handles service annotations on the controller and webhook services. ([#4329](https://github.com/jetstack/cert-manager/pull/4329), [@jwenz723](https://github.com/jwenz723))
 - Fixed a bug that can cause `cmctl version` to erroneously display the wrong webhook pod versions when older failed pods are present. ([#4615](https://github.com/jetstack/cert-manager/pull/4615), [@johnwchadwick](https://github.com/johnwchadwick))
 - Fixes a bug where a previous failed CertificateRequest was picked up during the next issuance. Thanks to @MattiasGees for raising the issue and help with debugging! ([#4688](https://github.com/jetstack/cert-manager/pull/4688), [@irbekrm](https://github.com/irbekrm))
-- Fixes an issue in `cmctl` that prevented displaying the Order resource with cert-manager
-  1.6 when running `cmctl status certificate`. ([#4569](https://github.com/jetstack/cert-manager/pull/4569), [@maelvls](https://github.com/maelvls))
 - Improve checksum validation in makefile based tool installation ([#4680](https://github.com/jetstack/cert-manager/pull/4680), [@SgtCoDFish](https://github.com/SgtCoDFish))
 - The HTTP-01 ACME solver now uses the `kubernetes.io/ingress.class` annotation instead of the `spec.ingressClassName` in created Ingress resources. ([#4762](https://github.com/jetstack/cert-manager/pull/4762), [@jakexks](https://github.com/jakexks))
 - The `cmctl experimental install` command now uses the cert-manager namespace. This fixes a bug which was introduced in release 1.6 that caused cert-manager to be installed in the default namespace. ([#4763](https://github.com/jetstack/cert-manager/pull/4763), [@wallrj](https://github.com/wallrj))
-- Update to latest version of keystore-go to address a backwards incompatible change introduced in v1.6.0 ([#4563](https://github.com/jetstack/cert-manager/pull/4563), [@SgtCoDFish](https://github.com/SgtCoDFish))
 
 ### Other (Cleanup or Flake)
 
 - Adds `clock_time_seconds_gauge` metric which returns the current clock time, based on seconds since 1970/01/01 UTC ([#4640](https://github.com/jetstack/cert-manager/pull/4640), [@JoshVanL](https://github.com/JoshVanL))
 - Adds an automated script for cert-manager developers to update versions of kind used for development and testing. ([#4574](https://github.com/jetstack/cert-manager/pull/4574), [@SgtCoDFish](https://github.com/SgtCoDFish))
-- Breaking change: removes the deprecated `dns01-self-check-nameservers` flag. Use `--dns01-recursive-nameservers` instead. ([#4551](https://github.com/jetstack/cert-manager/pull/4551), [@irbekrm](https://github.com/irbekrm))
 - Bump kind image versions ([#4593](https://github.com/jetstack/cert-manager/pull/4593), [@SgtCoDFish](https://github.com/SgtCoDFish))
 - Clean up: Remove `v1beta1` form the webhook's `admissionReviewVersions` as cert-manager no longer supports v1.16 ([#4639](https://github.com/jetstack/cert-manager/pull/4639), [@JoshVanL](https://github.com/JoshVanL))
 - Cleanup: Pipe feature gate flag to the e2e binary. Test against shared Feature Gate map for feature enabled and whether they should be tested against. ([#4703](https://github.com/jetstack/cert-manager/pull/4703), [@JoshVanL](https://github.com/JoshVanL))
@@ -89,7 +83,6 @@
 - github.com/opencontainers/image-spec: [v1.0.1 → v1.0.2](https://github.com/opencontainers/image-spec/compare/v1.0.1...v1.0.2)
 - github.com/opencontainers/runc: [v1.0.2 → v1.1.0](https://github.com/opencontainers/runc/compare/v1.0.2...v1.1.0)
 - github.com/opencontainers/selinux: [v1.9.1 → v1.10.0](https://github.com/opencontainers/selinux/compare/v1.9.1...v1.10.0)
-- github.com/pavel-v-chernykh/keystore-go/v4: [v4.1.0 → v4.2.0](https://github.com/pavel-v-chernykh/keystore-go/v4/compare/v4.1.0...v4.2.0)
 - github.com/prometheus/common: [v0.26.0 → v0.28.0](https://github.com/prometheus/common/compare/v0.26.0...v0.28.0)
 - github.com/seccomp/libseccomp-golang: [v0.9.1 → 3879420](https://github.com/seccomp/libseccomp-golang/compare/v0.9.1...3879420)
 - github.com/yuin/goldmark: [v1.3.5 → v1.4.0](https://github.com/yuin/goldmark/compare/v1.3.5...v1.4.0)
```

From what I understand, this weird behaviour comes from the fact that `release-notes`'s [`listCommits`](https://github.com/kubernetes/release/blob/50f63a508a03a173f934a95a0acbc3d4fb896d80/pkg/notes/notes.go#L520) is rather special: it first finds the committer date of `START_REV` and `START_END`, and then it iterates over `release-1.7` within this time period.

With `START_REV=v1.6.0`, the "start date" is 29 Oct 2021; conversely, with `START_REV=v1.6.0`, the "start date" is 15 Oct 2021, which explains why less entries are shown in `release-notes-v1.6.1.md` than in `release-notes-v1.6.0.md`.

Noticeably, the following "breaking change" doesn't show [in v1.7.0](https://github.com/jetstack/cert-manager/releases/tag/v1.7.0):

- Breaking change: removes the deprecated `dns01-self-check-nameservers` flag. Use `--dns01-recursive-nameservers` instead. ([#4551](https://github.com/jetstack/cert-manager/pull/4551), [@irbekrm](https://github.com/irbekrm))
- Breaking change: pprof now runs by default on localhost:6060 for webhook and controller, but only if explicitly enabled. Pprof can now be enabled also for cainjector. All three components have `--enable-profiling`, `--profiler-address` CLI flags to configure profiling. Thanks to @bitscuit for help with this! ([#4550](https://github.com/jetstack/cert-manager/pull/4550), [@irbekrm](https://github.com/irbekrm))
- Adds release targets for both `cmctl` as well as `kubectl-cert_manager` ([#4523](https://github.com/jetstack/cert-manager/pull/4523), [@JoshVanL](https://github.com/JoshVanL))
- Fixes an issue in `cmctl` that prevented displaying the Order resource with cert-manager 1.6 when running `cmctl status certificate`. ([#4569](https://github.com/jetstack/cert-manager/pull/4569), [@maelvls](https://github.com/maelvls))
- Update to latest version of keystore-go to address a backwards incompatible change introduced in v1.6.0 ([#4563](https://github.com/jetstack/cert-manager/pull/4563), [@SgtCoDFish](https://github.com/SgtCoDFish))

I think we should use `START_REV=v1.6.0` so that these entries show in the changelog entries; we should also add the above 5 items to https://cert-manager.io/docs/release-notes/release-notes-1.7/ and https://github.com/jetstack/cert-manager/releases/tag/v1.7.0.

Comment (maelvls) [reply content/en/docs/release-notes/release-notes-1.7.md:132]: I opened https://github.com/cert-manager/website/pull/830 with an explanation on which `START_REV` to be used when doing a final release

Comment (maelvls) [reply content/en/docs/release-notes/release-notes-1.7.md:132]: Update: in order to "skip" the commit that belongs to the previous release, I now run:

```sh
git fetch origin release-1.7
git rev-list --reverse --ancestry-path v1.6.0..release-1.7 | head -1
```

(inspired from [this StackOverflow response](https://stackoverflow.com/questions/2263674/how-do-i-find-the-next-commit-in-git-child-children-of-ref))

To generate the release notes for 1.7.0, I run:

```sh
git fetch origin release-1.7
GITHUB_TOKEN=$(lpass show -p github.com) \
START_SHA="$(git rev-list --reverse --ancestry-path v1.6.0..release-1.7 | head -1)" \
END_REV="v1.7.0" \
BRANCH="release-1.7" \
RELEASE_VERSION="1.7.0" \
release-notes --repo-path=cert-manager --org=cert-manager --repo=cert-manager --required-author=jetstack-bot --output release-notes.md
```

That gives the following diff:

```diff
--- mine/release-notes-1.7.0-from-1.6.1.md	2022-02-04 08:05:38.919191039 +0100
+++ mine/release-notes-1.7.0-from-1.6.0.md	2022-02-04 08:04:24.169352300 +0100
@@ -14,6 +14,7 @@
 - Added `additionalOutputFormats` parameter to allow `DER` (binary) and `CombinedPEM` (key + cert bundle) formats. ([#4598](https://github.com/cert-manager/cert-manager/pull/4598), [@seuf](https://github.com/seuf))
 - Added a makefile based build workflow which doesn't depend on bazel ([#4554](https://github.com/cert-manager/cert-manager/pull/4554), [@SgtCoDFish](https://github.com/SgtCoDFish))
 - Added a new Helm chart parameter `prometheus.servicemonitor.honorLabels`, which sets the `honor_labels` field  of the Prometheus scrape config. ([#4608](https://github.com/cert-manager/cert-manager/pull/4608), [@thirdeyenick](https://github.com/thirdeyenick))
+- Breaking change: pprof now runs by default on localhost:6060 for webhook and controller, but only if explicitly enabled. Pprof can now be enabled also for cainjector. All three components have `--enable-profiling`, `--profiler-address` CLI flags to configure profiling. Thanks to @bitscuit for help with this! ([#4550](https://github.com/cert-manager/cert-manager/pull/4550), [@irbekrm](https://github.com/irbekrm))
 - Certificate Secrets are now managed by the APPLY API call, rather than UPDATE/CREATE. The issuing controller actively reconciles Certificate SecretTemplate's against corresponding Secrets, garbage collecting and correcting key/value changes. ([#4638](https://github.com/cert-manager/cert-manager/pull/4638), [@JoshVanL](https://github.com/JoshVanL))
 
 ### Bug or Regression
@@ -23,14 +24,18 @@
 - Fixed a bug in the way the Helm chart handles service annotations on the controller and webhook services. ([#4329](https://github.com/cert-manager/cert-manager/pull/4329), [@jwenz723](https://github.com/jwenz723))
 - Fixed a bug that can cause `cmctl version` to erroneously display the wrong webhook pod versions when older failed pods are present. ([#4615](https://github.com/cert-manager/cert-manager/pull/4615), [@johnwchadwick](https://github.com/johnwchadwick))
 - Fixes a bug where a previous failed CertificateRequest was picked up during the next issuance. Thanks to @MattiasGees for raising the issue and help with debugging! ([#4688](https://github.com/cert-manager/cert-manager/pull/4688), [@irbekrm](https://github.com/irbekrm))
+- Fixes an issue in `cmctl` that prevented displaying the Order resource with cert-manager
+  1.6 when running `cmctl status certificate`. ([#4569](https://github.com/cert-manager/cert-manager/pull/4569), [@maelvls](https://github.com/maelvls))
 - Improve checksum validation in makefile based tool installation ([#4680](https://github.com/cert-manager/cert-manager/pull/4680), [@SgtCoDFish](https://github.com/SgtCoDFish))
 - The HTTP-01 ACME solver now uses the `kubernetes.io/ingress.class` annotation instead of the `spec.ingressClassName` in created Ingress resources. ([#4762](https://github.com/cert-manager/cert-manager/pull/4762), [@jakexks](https://github.com/jakexks))
 - The `cmctl experimental install` command now uses the cert-manager namespace. This fixes a bug which was introduced in release 1.6 that caused cert-manager to be installed in the default namespace. ([#4763](https://github.com/cert-manager/cert-manager/pull/4763), [@wallrj](https://github.com/wallrj))
+- Update to latest version of keystore-go to address a backwards incompatible change introduced in v1.6.0 ([#4563](https://github.com/cert-manager/cert-manager/pull/4563), [@SgtCoDFish](https://github.com/SgtCoDFish))
 
 ### Other (Cleanup or Flake)
 
 - Adds `clock_time_seconds_gauge` metric which returns the current clock time, based on seconds since 1970/01/01 UTC ([#4640](https://github.com/cert-manager/cert-manager/pull/4640), [@JoshVanL](https://github.com/JoshVanL))
 - Adds an automated script for cert-manager developers to update versions of kind used for development and testing. ([#4574](https://github.com/cert-manager/cert-manager/pull/4574), [@SgtCoDFish](https://github.com/SgtCoDFish))
+- Breaking change: removes the deprecated `dns01-self-check-nameservers` flag. Use `--dns01-recursive-nameservers` instead. ([#4551](https://github.com/cert-manager/cert-manager/pull/4551), [@irbekrm](https://github.com/irbekrm))
 - Bump kind image versions ([#4593](https://github.com/cert-manager/cert-manager/pull/4593), [@SgtCoDFish](https://github.com/SgtCoDFish))
 - Clean up: Remove `v1beta1` form the webhook's `admissionReviewVersions` as cert-manager no longer supports v1.16 ([#4639](https://github.com/cert-manager/cert-manager/pull/4639), [@JoshVanL](https://github.com/JoshVanL))
 - Cleanup: Pipe feature gate flag to the e2e binary. Test against shared Feature Gate map for feature enabled and whether they should be tested against. ([#4703](https://github.com/cert-manager/cert-manager/pull/4703), [@JoshVanL](https://github.com/JoshVanL))
@@ -83,6 +88,7 @@
 - github.com/opencontainers/image-spec: [v1.0.1 → v1.0.2](https://github.com/opencontainers/image-spec/compare/v1.0.1...v1.0.2)
 - github.com/opencontainers/runc: [v1.0.2 → v1.1.0](https://github.com/opencontainers/runc/compare/v1.0.2...v1.1.0)
 - github.com/opencontainers/selinux: [v1.9.1 → v1.10.0](https://github.com/opencontainers/selinux/compare/v1.9.1...v1.10.0)
+- github.com/pavel-v-chernykh/keystore-go/v4: [v4.1.0 → v4.2.0](https://github.com/pavel-v-chernykh/keystore-go/v4/compare/v4.1.0...v4.2.0)
 - github.com/prometheus/common: [v0.26.0 → v0.28.0](https://github.com/prometheus/common/compare/v0.26.0...v0.28.0)
 - github.com/seccomp/libseccomp-golang: [v0.9.1 → 3879420](https://github.com/seccomp/libseccomp-golang/compare/v0.9.1...3879420)
 - github.com/yuin/goldmark: [v1.3.5 → v1.4.0](https://github.com/yuin/goldmark/compare/v1.3.5...v1.4.0)
@@ -123,4 +129,3 @@
```

I'll add the "git rev-list --reverse" trick to the release process.

**Comment by maelvls** [comment]
I went ahead and edited the GitHub Release description for [v1.7.0](https://github.com/cert-manager/cert-manager/releases/tag/v1.7.0) with the missing 4 changelog entries.

**Review by wallrj — CHANGES_REQUESTED** [review]
Thanks @maelvls 

Great detective work sorting out the correct start and end revisions for the release-notes tool and thanks for tidying everything up.

Only thing I want changed is that you remove the youtube link. 
I think it is likely to confuse people. 

/hold

**Review by wallrj — APPROVED** [review]
/lgtm
/unhold

**Pull request merged**
